js/common/modules/@arangodb/graph/graphs-generation.js

```diff
@@ -0,0 +1,193 @@
+            ++firstFree;
+            // the last level of vertices is not collected to leavesNestLevel: its vertices have no children
+            if (d <= treeDepth - 2) {
+                leavesNestLevel.push(firstFree - 2);
+                leavesNestLevel.push(firstFree - 1);
+            }
+            leaves = leavesNestLevel;
+        }
+
+    }
+    return {vertices, edges};
+}
+
+// Creates a graph with size many vertices such that for each pair (v,w) of distinct vertices,
+// (v,w) or (w,v) is an edge (or both). There are no self-loops.
+// The parameter kind has the following meaning:
+//  - "bidirected": for all distinct vertices v,w, there are edges (v,w) and (w,v)
+//  - "linear": the edges constitute a linear order <:
+//              there is an edge from v to w if and only if index of v < index of w.
+// Note that the number of edges grows quadratically in size!
+function createClique(size, vColl, name_prefix, kind = "bidirected") {
+    let vertices = makeVertices(size, name_prefix);
+    let edges = [];
+    for (let v = 0; v < size; ++v) {
+        for (let w = v + 1; w < size; ++w) {
+            switch (kind) {
+                case "bidirected":
+                    edges.push(makeEdge(v, w, vColl, name_prefix));
+                    edges.push(makeEdge(w, v, vColl, name_prefix));
+                    break;
+                case "linear":
+                    edges.push(makeEdge(v, w, vColl, name_prefix));
+                    break;
+            }
+        }
+    }
+    return {vertices, edges};
+}
+
+// for manual testing
+
+// console.log(createDirectedCycle(2, "vertices", "v"));
+// console.log(createDirectedCycle(3, "vertices", "v"));
+// console.log(createAlternatingCycle(2, "vertices", "v"));
+// console.log(createAlternatingCycle(3, "vertices", "v"));
+// console.log(createAlternatingCycle(4, "vertices", "v"));
+// console.log(createFullBinaryTree(0, "vertices", "v", false));
+// console.log(createFullBinaryTree(0, "vertices", "v", false));
+// console.log(createFullBinaryTree(0, "vertices", "v", true));
+// console.log(createFullBinaryTree(1, "vertices", "v", false));
+// console.log(createFullBinaryTree(1, "vertices", "v", true));
+// console.log(createFullBinaryTree(2, "vertices", "v", false));
+// console.log(createFullBinaryTree(2, "vertices", "v", true));
+// console.log(createFullBinaryTree(3, "vertices", "v", false));
+// console.log(createFullBinaryTree(3, "vertices", "v", true));
+// console.log(createClique(0, "vertices", "v", "bidirected"));
+// console.log(createClique(1, "vertices", "v", "bidirected"));
+// console.log(createClique(2, "vertices", "v", "bidirected"));
+// console.log(createClique(3, "vertices", "v", "bidirected"));
+// console.log(createClique(1, "vertices", "v", "linear"));
+// console.log(createClique(2, "vertices", "v", "linear"));
+// console.log(createClique(3, "vertices", "v", "linear"));
+// console.log(createClique(4, "vertices", "v", "linear"));
```

Comment:
I am not sure whether this is considered to be documentation, and if so, whether its very useful as that;
For manual testing you can require this module in an arangosh and execute the create functions.

Reply:
This is temporary as a working tool, it should be removed in this form when the PR is ready for merging. I am going to add some tests. (Actually, this PR should be a draft, how did it change to "open"?)